### PR TITLE
fix particle system alpha and lighting for fixed pipeline

### PR DESCRIPTION
**Pull request by mp3butcher**
same as https://github.com/OpenMW/openmw/pull/2441 but only for FFP (easier to review)
it fixes the alpha issue https://gitlab.com/OpenMW/openmw/issues/3676

Capture d’écran de 2019-07-11 03-42-04

### CODE REVIEW

**Comment by akortunov** [comment]
Just for reference: it works only in the fixed pipeline mode.

**Comment by Capostrophic** [comment]
Particle lighting doesn't behave right in shader pipeline, there's a bug filed for that. This seems to be worth looking into since there's no bug in osgParticle itself.

**Comment by mp3butcher** [comment]
@akortunov 
...that's the reason of the 2 differents prs...Making a simplier one (fixing only fixed pipeline) intend not to be overhelming for review

**Comment by AnyOldName3** [comment]
Can you explain why the changes here help?

**Comment by mp3butcher** [comment]
@AnyOldName3 
commit1 : disable lighting because material cause strange behavior with gl_Color alpha
commit2: separate function over time for Alpha and Color

**Comment by AnyOldName3** [comment]
Can you be clear and specific about that, please? I've done some investigation into OpenGL material colours and how they interact with vertex colours in the past, so I might be able to find a better way of sorting that out without disabling lighting if you can tell me exactly what you've found, and I don't know how the colour and alpha are used in `osgParticle` exactly, so you might be able to enlighten me much faster than I could look through the source code.

**Comment by mp3butcher** [comment]
Material seems to affect gl_Color preventing alpha control of particlesystem...and disabling material is not enough to prevent that, so i disabled light. (not a problem particlesystems have its own renderbin)
Perhaps it's linked to some normal bore effect as particles don't have normals (and so can't be used for lighting)
Don't know

**Comment by AnyOldName3** [comment]
You know about `glColorMaterial` and what it does, right?

**Comment by mp3butcher** [comment]
please... I know how to disable Material...

**Comment by AnyOldName3** [comment]
Just checking - not everyone still knows how the FFP works.

**Comment by mp3butcher** [comment]
i found that scrawl commit https://github.com/OpenMW/openmw/commit/0a411cbd7cd55cacb5c8c2c1efcbf5dddf8f862e#diff-d913374ea75d1e583d8c7a5d028e4f4f
but it seems this code was removed 
it can explain shy i have to disable lighting

**Comment by mp3butcher** [comment]
seems the proper fix in fact

**Comment by Capostrophic** [comment]
Pretty sure just removing that code causes severe rendering glitches with particles.

**Comment by mp3butcher** [comment]
@Capostrophic 
give argument against this change not your assurance...you're never sure....

**Comment by mp3butcher** [comment]
reopened just for a win32 test

File: components/nifosg/particle.cpp [hunk 1]
```diff
@@ -191,6 +191,7 @@ void ParticleColorAffector::operate(osgParticle::Particle* particle, double /* d
 
     particle->setColorRange(osgParticle::rangev4(color, color));
     particle->setAlphaRange(osgParticle::rangef(alpha, alpha));
+
```
Comment (akortunov) [inline components/nifosg/particle.cpp:194]: Please discard this redundant change and squash commits.

Comment (mp3butcher) [reply components/nifosg/particle.cpp:194]: > 
> 
> reopened just for a win32 test
I'll clean that pr as test is pretty satisfying..don't know why i closed it

Comment (mp3butcher) [reply components/nifosg/particle.cpp:194]: ah yes i remember, someone nags me about stuff that wasn't in nif

**Pull request closed without merging**